stubs/paramiko/paramiko/channel.pyi

```diff
@@ -41,14 +41,14 @@ class Channel(ClosingContextManager):
     def __init__(self, chanid: int) -> None: ...
     def __del__(self) -> None: ...
     def get_pty(
-        self, term: str = ..., width: int = ..., height: int = ..., width_pixels: int = ..., height_pixels: int = ...
+        self, term: bytes = ..., width: int = ..., height: int = ..., width_pixels: int = ..., height_pixels: int = ...
     ) -> None: ...
     def invoke_shell(self) -> None: ...
     def exec_command(self, command: str) -> None: ...
     def invoke_subsystem(self, subsystem: str) -> None: ...
     def resize_pty(self, width: int = ..., height: int = ..., width_pixels: int = ..., height_pixels: int = ...) -> None: ...
-    def update_environment(self, environment: Mapping[str, str]) -> None: ...
-    def set_environment_variable(self, name: str, value: str) -> None: ...
+    def update_environment(self, environment: Mapping[bytes, bytes]) -> None: ...
+    def set_environment_variable(self, name: bytes, value: bytes) -> None: ...
```

Comment:
Same here for both `name` and `value`.

Reply:
Thanks for the information.

The arguments of ``paramiko.ServerInterface.check_channel_env_request`` are `bytes`. 

In my application the arguments are typechecked with typeguard and the types should be correct.

https://github.com/ssh-mitm/ssh-mitm/blob/8bffa3b8e309887b8729a9fede7f233f7c85f4e2/ssh_proxy_server/interfaces/server.py#L265:L266

The received arguments are passed to `paramiko.Channel.set_environment_variable`. So there is a difference in the argument types of both methods.

Perhaps the this was my mistake and I assumed, that both methods accept the same types.

I think this needs further tests. I will check that.

Reply:
`paramiko.Channel.set_environment_variable` creates a new message and adds the name and the vallue with the method add_string.

https://github.com/paramiko/paramiko/blob/a598ca14d6451343c69be97e0e1c65eb15c0e607/paramiko/channel.py#L360:L361

The method add_string converts the string to bytes: https://github.com/paramiko/paramiko/blob/a598ca14d6451343c69be97e0e1c65eb15c0e607/paramiko/message.py#L273

To convert the value to bytes, the method asbytes is used: https://github.com/paramiko/paramiko/blob/a598ca14d6451343c69be97e0e1c65eb15c0e607/paramiko/common.py#L193

asbytes uses the function `b`: https://github.com/paramiko/paramiko/blob/a598ca14d6451343c69be97e0e1c65eb15c0e607/paramiko/py3compat.py#L135

The function `b` is able to use `str` and `bytes`.

The same is true for `get_pty`.

**So a lot of types have to be changed to `str | bytes`** :disappointed_relieved:

Reply:
Don't feel obligated to change anything that you don't need. Any gradual improvement is welcome!

Reply:
:+1: